CMakeLists.txt

```diff
@@ -9,6 +9,37 @@ project(Ship VERSION 8.0.5 LANGUAGES C CXX)
 set(PROJECT_BUILD_NAME "MacReady Foxtrot" CACHE STRING "")
```

Comment:
i've been hoping to be able to remove the hardcoded version number and version name, maybe it'd be possible to pull these from the tag too (i've generally made sure that tags have the named version as the description)

the question then becomes what do we do for builds that aren't releases - we'll still need to set a version number, maybe for the name we could just have "MacReady" for the named release branch and then "Nightly" or "Develop" for the main dev branch

in general i'd be very happy to see the extra info make it into the builds, i just want to make sure we think through all of the use cases we want to support

Reply:
> the question then becomes what do we do for builds that aren't releases - we'll still need to set a version number, maybe for the name we could just have "MacReady" for the named release branch and then "Nightly" or "Develop" for the main dev branch

Do you mean for it to be just `MacReady`/`Develop` with no second part?
If so we would still need to have `MacReady` somewhere in there as I see no way of getting it from the branch reliably.

Reply:
so i'm not quite sure what we'd want to set those values to when not on a tagged version, but i'm thinking we could just not *show* them on the logo screen when not on a tagged version

i'm still really hoping we can figure out a way to use the git tag description to set the version name for tagged versions instead of needing to hardcode it, it seems we can get that by using
```sh
git cat-file tag <tag_version> | sed '1,/^$/d'
```
but in my quick poking around i wasn't able to get that working in cmake

basically i'm hoping we can do

```cmake
project(Ship VERSION "${GIT_COMMIT_TAG}" LANGUAGES C CXX)
set(PROJECT_BUILD_NAME "${GIT_TAG_DESCRIPTION}" CACHE STRING "")
```
when we have a tag, and then maybe just do
```cmake
project(Ship VERSION 0.0.0 LANGUAGES C CXX)
set(PROJECT_BUILD_NAME "Unreleased" CACHE STRING "")
```
when we don't have a tag?

Reply:
I'm a bit hesitant on `0.0.0` since that version info is what gets stored into the game otrs and is used for enforcing the regenerate flow. I know what we have isn't great either since the develop branch wont complain about an otr from the current release.